ratatui-widgets/src/mascot.rs

```diff
@@ -0,0 +1,188 @@
+              hhhhhhhhh
+             hhhhhhhhhh
+            hhhhhhhhhhhh
+            hhhhhhhhhhhhh
+            hhhhhhhhhhhhh     ██████
+             hhhhhhhhhhh    ████████
+                  hhhhh ███████████
+                   hhh ██ee████████
+                    h █████████████
+                ████ █████████████
+               █████████████████
+               ████████████████
+               ████████████████
+                ███ ██████████
+              ▒▒    █████████
+             ▒░░▒   █████████
+            ▒░░░░▒ ██████████
+           ▒░░▓░░░▒ █████████
+          ▒░░▓▓░░░░▒ ████████
+         ▒░░░░░░░░░░▒ ██████████
+        ▒░░░░░░░░░░░░▒ ██████████
+       ▒░░░░░░░▓▓░░░░░▒ █████████
+      ▒░░░░░░░░░▓▓░░░░░▒ ████  ███
+     ▒░░░░░░░░░░░░░░░░░░▒ ██   ███
+    ▒░░░░░░░░░░░░░░░░░░░░▒ █   ███
+    ▒░░░░░░░░░░░░░░░░░░░░░▒   ███
+     ▒░░░░░░░░░░░░░░░░░░░░░▒ ███
+      ▒░░░░░░░░░░░░░░░░░░░░░▒ █"
+};
+
+const TERM: char = '░';
+const TERM_BORDER: char = '▒';
+
+/// State for the mascot's eye
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum MascotEye {
+    /// The default eye color
+    #[default]
+    Default,
+
+    /// The red eye color
+    Red,
+}
+
+/// A widget that renders the Ratatui mascot
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct RatatuiMascot {
+    eye_state: MascotEye,
+    /// The color of the rat
+    rat_color: Color,
+    /// The color of the rat's eye
+    rat_eye_color: Color,
+    /// The color of the rat's eye when blinking
+    rat_eye_blink: Color,
+    /// The color of the rat's hat
+    hat_color: Color,
+    /// The color of the terminal
+    term_color: Color,
+    /// The color of the terminal border
+    term_border_color: Color,
+    /// The color of the terminal cursor
+    term_cursor_color: Color,
+}
+
+impl Default for RatatuiMascot {
+    fn default() -> Self {
+        Self {
+            rat_color: Color::Indexed(252),         // light_gray #d0d0d0
+            hat_color: Color::Indexed(231),         // white #ffffff
+            rat_eye_color: Color::Indexed(236),     // dark_charcoal #303030
+            rat_eye_blink: Color::Indexed(196),     // red #ff0000
+            term_color: Color::Indexed(232),        // vampire_black #080808
+            term_border_color: Color::Indexed(235), // raisin_black  #262626
+            term_cursor_color: Color::Indexed(248), // dark_gray #a8a8a8
+            eye_state: MascotEye::Default,
+        }
+    }
+}
+
+impl RatatuiMascot {
+    /// Create a new Ratatui mascot widget
+    pub fn new() -> Self {
+        Self {
+            ..Default::default()
+        }
+    }
+
+    /// Tiny help to set the color of the rat's eye
+    #[must_use]
+    pub const fn set_eye(self, rat_eye: MascotEye) -> Self {
+        Self {
+            eye_state: rat_eye,
+            ..self
+        }
+    }
+    const fn color_for(&self, c: char) -> Option<Color> {
+        match c {
+            '█' => Some(self.rat_color),
```

Comment:
A few constants for these characters would be a good idea here.

Reply:
(btw. Make sure to include the space as it will make the match statement neater)